recipes/spdlog/all/conanfile.py

```diff
@@ -25,6 +25,7 @@ class SpdlogConan(ConanFile):
         "wchar_support": [True, False],
         "wchar_filenames": [True, False],
         "no_exceptions": [True, False],
+        "fmt_header_only": [True, False],
```

Comment:
I disagree with this new option, it's a problem which must be solved by conan client

Reply:
While I do see your point, here this change is very specific to this recipe and this package, as it is an option provided by the author of the library, not (directly) a way to circumvent a Conan limitation.

If you didn't use Conan you'd **have** to specify this manually by giving cmake a prefix path or module path or whatever path to an external fmt, and this PR makes Conan do exactly that.

If Conan cannot do this automatically now and neither for a (long) while, then this change makes sense by not burdening each and every consumer of this package configuration by forcing a custom build or a global configuration that affects the whole way Conan resolves packages.

/s Deleting this option in the future is also possible, as many recipes have done in the past and breaking the world with it (e.g. nlohmann_json). /end sarcasm

Or we could silently ignore it as a temporary deprecation mechanism. If Conan cannot solve this problem now or for a long time, leaving this package in this state does not seem like a good compromise IMO (and not just because it breaks my personal and professional workflow).

Reply:
> While I do see your point, here this change is very specific to this recipe and this package, as it is an option provided by the author of the library, not (directly) a way to circumvent a Conan limitation.

It doesn't matter, this spdlog option is an internal CMake detail to handle the fact that CMake imported target of fmt is different when fmt is header only, there is no other reason, no deep logic inside spdlog depending on fmt header only, static or shared.
It's the kind of option you could see in many libraries if some guy would have decided to provide `ZLIB::ZLIB_static` & `ZLIB::ZLIB_shared` in `FindZLIB.cmake` instead of just `ZLIB::ZLIB`.